homeassistant/components/recorder/migration.py

```diff
@@ -549,7 +549,7 @@ def _apply_update(  # noqa: C901
     if dialect == SupportedDialect.MYSQL:
         timestamp_type = "DOUBLE PRECISION"
         context_bin_type = f"BLOB({CONTEXT_ID_BIN_MAX_LENGTH})"
-    if dialect == SupportedDialect.POSTGRESQL:
+    elif dialect == SupportedDialect.POSTGRESQL:
```

Comment:
My history actually was messed up after I upgraded from schema 30 to 41. The result was weird graphs and unresponsive ui.

Reply:
> My history actually was messed up after I upgraded from schema 30 to 41. The result was weird graphs and unresponsive ui.

This wasn't the cause of that as this is the code in dev
https://github.com/home-assistant/core/blob/b6d001bfe6f00db0461861c0894d5bc83306b038/homeassistant/components/recorder/migration.py#L525

Reply:
> just an inherent risk for those installing from dev. Since this didn't get released yet, it may be good enough to fix it now.

We generally don't do schema updates to fix `dev` only bugs.

> Will this need to have a schema bump, in order to resolve a broken timestamp schema for anyone who may have already performed a dev install with MySQL?

It would have only happened if imported an older database into `dev` and didn't setup a fresh environment.

Reply:
> > My history actually was messed up after I upgraded from schema 30 to 41. The result was weird graphs and unresponsive ui.
> 
> Unless you are running `dev` in production (happy to have the additional testing), this wasn't the cause of that as this is the code in 2023.3.x
> 
> https://github.com/home-assistant/core/blob/b6d001bfe6f00db0461861c0894d5bc83306b038/homeassistant/components/recorder/migration.py#L525

That commit has schema version 35. If @dcoder42's issue was introduced when upgrading to schema 41, that was only bumped a week ago in https://github.com/home-assistant/core/commit/dbc0890ce8844143d1984bc46b9a14840515ea7f. So it was in dev.

> We generally don't do schema updates to fix dev only bugs. If it makes it to beta without being caught that is generally when we will do another schema bump.

Got it, makes sense.

Reply:
Also we have automatic repair of stats schema precision in the code to handle it going wrong from manual SQLite to MariaDB migrations so that will fix itself.

We don't current do that for the events or states table though

Reply:
I'll look into breaking out the existing stats auto repair into its own module so we can add support for auto repairing events and states since it comes up enough from bad manual migrations in production branches